docs/website/docs/building-from-source/riscv.md

```diff
@@ -0,0 +1,137 @@
+# RISC-V cross-compilation
+
+Running on a platform like RISC-V involves cross-compiling from a _host_
+platform (e.g. Linux) to a _target_ platform (a specific RISC-V CPU architecture
+and operating system):
+
+* IREE's _compiler_ is built on the host and is used there to generate modules
+  for the target
+* IREE's _runtime_ is built on the host for the target. The runtime is then
+  pushed to the target to run natively.
+
+## Prerequisites
+
+### Host environment setup
+
+You should already be able to build IREE from source on your host platform.
+Please make sure you have followed the [getting started](./getting-started.md)
+steps.
+
+### Install RISC-V cross-compile toolchain and emulator
+
+You'll need a RISC-V LLVM compilation toolchain and a RISC-V enabled QEMU
+emulator.
+
+* RISC-V toolchain is built from https://github.com/llvm/llvm-project (main branch).<br>
+  * Currently, the LLVM compiler is built on GNU toolchain, including libgcc,
+    GNU linker, and C libraries. You need to build GNU toolchain first.<br>
+  * Clone GNU toolchain from: https://github.com/riscv/riscv-gnu-toolchain
+    (master branch). Switch the "riscv-binutils" submodule to `rvv-1.0.x-zfh`
+    branch manually.
+* RISC-V QEMU is built from https://github.com/sifive/qemu/tree/v5.2.0-rvv-rvb-zfh
+
+An environment variable `RISCV_TOOLCHAIN_ROOT` needs
+to be set to the root directory of the installed GNU toolchain. The variable can
+be used in building the RISCV target and a LLVM AOT module.
+
+### Install Prebuilt RISC-V Tools (RISC-V 64-bit Linux toolchain)
+
+Execute the following script to download the prebuilt RISC-V toolchain and QEMU:
+
+```shell
+# In IREE source root
+$ ./build_tools/riscv/riscv_bootstrap.sh
+```
```

Comment:
Can you move the comment out of this code block and drop the leading `$`? (same for other code blocks)

---

I had been recommending this style (`# comments` and `$ echo "sample command"`) on the old documentation, but reversed my position on that with the new website.

The main reason is that mkdocs offers a "Copy to clipboard" button for code snippets that does not strip comments or leading prompt characters.

![image](https://user-images.githubusercontent.com/4010439/124155166-0b2f8c80-da4b-11eb-8032-eba5492b4194.png)

See also https://developers.google.com/style/code-syntax#prompt that _does_ recommend the leading `$`, but that website uses a widget that separates the code from the prompt: 
![image](https://user-images.githubusercontent.com/4010439/124156201-2fd83400-da4c-11eb-8112-6f8664932419.png)

What I really want is g3doc's feature set, but this isn't so bad.

Reply:
Done.